pip_audit/_dependency_source/resolvelib/resolvelib.py

```diff
@@ -53,12 +60,18 @@ def __init__(
 
         `state` is an `AuditState` to use for state callbacks.
         """
-        self.provider = PyPIProvider(index_urls, timeout, cache_dir, state)
+        self.index_urls = index_urls
+        self.timeout = timeout
+        self.session = caching_session(cache_dir, use_pip=True)
```

Comment:
I'm creating the `session` here so I can patch it in the tests. It's harder to keep it in the `PyPIProvider` because those get created on every `resolve` call.

Reply:
Works for me -- IMO it's worth a comment explaining that we're doing it to make testing a little easier, but `requests` sessions are self-managing so I have no objections 🙂

Reply:
[aae5e80](https://github.com/pypa/pip-audit/pull/462/commits/aae5e80695526e85a1831c5e4b4164e7cf1c256c)